src/watchmaker/workers/yum.py

```diff
@@ -54,40 +41,15 @@ def _get_amazon_el_version(self, version):
 
     def get_dist_info(self):
         """Validate the Linux distro and return info about the distribution."""
-        dist = None
-        version = None
+        dist = distro.id()
+        version = distro.version()[0]
         el_version = None
 
-        # Read first line from /etc/system-release
-        try:
-            with io.open('/etc/system-release', encoding='utf8') as fh_:
-                release = fh_.readline().strip()
-        except Exception:
-            self.log.critical(
-                'Failed to read /etc/system-release. Cannot determine system '
-                'distribution!'
-            )
-            raise
-
-        # Search the release file for a match against _supported_dists
-        matched = self.DIST_PATTERN.search(release.lower())
-        if matched is None:
-            # Release not supported, exit with error
-            msg = (
-                'Unsupported OS distribution. OS must be one of: {0}'
-                .format(', '.join(self.SUPPORTED_DISTS))
-            )
-            self.log.critical(msg)
-            raise WatchmakerError(msg)
-
-        # Assign dist,version from the match groups tuple, removing any spaces
-        dist, version = (x.replace(' ', '') for x in matched.groups())
```

Comment:
Ahh, lol, _this_ is what was changing `red hat` to `redhat`. Ok, now that we're using `distro`, I think we can just re-map `rhel` to `redhat` explicitly?

Reply:
Err.. is that outdated code-content? Doing a search in my editor for `x.replace` gives me:
```
E486: Pattern not found: x.replace
```

Reply:
That makes sense, since the comment is on a removed line. ;)

It was intended as providing further details and context on this comment, https://github.com/plus3it/watchmaker/pull/2389/files#r1023345795

Reply:
Ok, so… Resolve?

Reply:
I think we'll be better off creating a map of `distro.id()` values to watchmaker "config" values. That would mean something like changing `SUPPORTED_DISTS` to be a map:

```
    SUPPORTED_DISTS = {
        'amazon': 'amazon',
        'centos': 'centos',
        'rhel': 'redhat'
    }
```

and then we lookup the dist from that map:

```
        try:
            dist = self.SUPPORTED_DISTS[distro.id()]
        except KeyError:
            # Release not supported, exit with error
            msg = (
                'Unsupported OS distribution. OS must be one of: {0}'
                .format(', '.join(self.SUPPORTED_DISTS.keys()))
            )
            self.log.critical(msg)
            raise WatchmakerError(msg)
```

Reply:
👍 

Cool. I'll give that a shot – possibly as early as later this afternoon or early this evening.

Reply:
Ok, so, we're initially defining `dist` as the string returned by `distro.id()` (at line #72), but subsequently redefining it with the `try`/`except` block. Is that what we actually want to do? Or do we want to replace the current:

~~~
    def get_dist_info(self):
        """Validate the Linux distro and return info about the distribution."""
        dist = distro.id()
        version = distro.version()[0]
        el_version = None

        # Determine el_version
        if dist == 'amazon':
            el_version = self._get_amazon_el_version(version)
        else:
            el_version = distro.version()[0]

        if el_version is None:
            msg = (
                'Unsupported OS version! dist = {0}, version = {1}.'
                .format(dist, version)
            )
            self.log.critical(msg)
            raise WatchmakerError(msg)

        # Error if 'dist' is not found in SUPPORTED_DISTS
        if dist not in self.SUPPORTED_DISTS:
            # Release not supported, exit with error
            msg = (
                'Unsupported OS distribution. OS must be one of: {0}'
                .format(', '.join(self.SUPPORTED_DISTS))
            )
            self.log.critical(msg)
            raise WatchmakerError(msg)

        dist_info = {
            'dist': dist,
            'el_version': el_version
        }
        self.log.debug('dist_info=%s', dist_info)
        return dist_info
~~~

With something more like:

~~~
    def get_dist_info(self):
        """Validate the Linux distro and return info about the distribution."""
        # Error if 'dist' is not found in SUPPORTED_DISTS
        try:
            dist = self.SUPPORTED_DISTS[distro.id()]
        except KeyError:
            # Release not supported, exit with error
            msg = (
                'Unsupported OS distribution. OS must be one of: {0}'
                .format(', '.join(self.SUPPORTED_DISTS.keys()))
            )
            self.log.critical(msg)
            raise WatchmakerError(msg)
        version = distro.version()[0]
        el_version = None

        # Determine el_version
        if dist == 'amazon':
            el_version = self._get_amazon_el_version(version)
        else:
            el_version = distro.version()[0]

        if el_version is None:
            msg = (
                'Unsupported OS version! dist = {0}, version = {1}.'
                .format(dist, version)
            )
            self.log.critical(msg)
            raise WatchmakerError(msg)

        dist_info = {
            'dist': dist,
            'el_version': el_version
        }
        self.log.debug('dist_info=%s', dist_info)
        return dist_info
~~~

I'm not generally a fan of overloading variables, so, I'll implement the latter and see what breaks.

Reply:
Alright. The latter works.

Reply:
Indeed, my suggestion was targeted at replacing the initial assignment of `dist = distro.id()`.